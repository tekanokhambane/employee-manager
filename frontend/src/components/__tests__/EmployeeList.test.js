import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { createPinia } from 'pinia'
import EmplpoyeeList from '../EmployeeList.vue'
import EmployeeItemVue from '../EmployeeItem.vue'
import useEmployeesStore from '../../stores/employeesStore'

describe('EmployeeList.vue Test', () => {
    it('should render correctly', () => {
        const wrapper = shallowMount(EmplpoyeeList, {
            global: {
                plugins: [createPinia()]
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render child components', () => {
        const wrapper = shallowMount(EmplpoyeeList, {
            global: {
                plugins: [createPinia()]
            }
        })
        expect(wrapper.findComponent(EmployeeItemVue)).toBeTruthy()
    })

    it('should render child components', () => {
        const wrapper = shallowMount(EmplpoyeeList, {
            global: {
                plugins: [createPinia()]
            }
        })
        expect(wrapper.findComponent(EmployeeItemVue).exists())
    })



})

