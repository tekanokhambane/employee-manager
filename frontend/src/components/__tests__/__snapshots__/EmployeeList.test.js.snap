// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmployeeList.vue Test > should render correctly 1`] = `
"<!--v-if-->
<div data-v-d6d1597d="" class="employeeList">
  <div data-v-d6d1597d="" class="noItems"><img data-v-d6d1597d="" src="/src/assets/Icon.JPG" alt="No Items">
    <h2 data-v-d6d1597d="">There is nothing here</h2>
    <p data-v-d6d1597d="">Create a new employee by clicking the <span data-v-d6d1597d="">New Employee</span> button to get started</p>
  </div>
</div>"
`;

exports[`EmployeeList.vue Test > should render correctly 2`] = `
"<!--v-if-->
<div data-v-d6d1597d="" class="employeeList">
  <div data-v-d6d1597d="" class="noItems"><img data-v-d6d1597d="" src="/src/assets/Icon.JPG" alt="No Items">
    <h2 data-v-d6d1597d="">There is nothing here</h2>
    <p data-v-d6d1597d="">Create a new employee by clicking the <span data-v-d6d1597d="">New Employee</span> button to get started</p>
  </div>
</div>"
`;
